sql: implement the ALTER TABLE ... SET (...) syntax

otan opened the pull request:
see individual commits for details.
prep work for TTL.

otan committed:
paramparse: add pgcodes to certain error codes

Release note (sql change): Added pgcodes to errors when an invalid
storage parameter is passed.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/75262)
<!-- Reviewable:end -->

otan committed:
sql: implement the ALTER TABLE ... SET (...) syntax

Release note (sql change): Implemented the `ALTER TABLE ... SET (...)`
syntax. This mostly no-ops as we do not support any storage parameters
(yet).

rafiss reviewed (approved):
> Reviewed 3 of 3 files at r1, 8 of 8 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75262)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)



<!-- Sent from Reviewable.io -->

otan commented:
bors r=rafiss
